terraform/cp4ba/variables.tf

```diff
@@ -0,0 +1,232 @@
+//  }
+}
+
+variable "iaas_classic_api_key" {}
+variable "iaas_classic_username" {}
+variable "ssh_public_key_file" {}
+variable "ssh_private_key_file" {}
+variable "classic_datacenter" {}
+
+variable "config_dir" {
+  default     = "./.kube/config"
+  description = "directory to store the kubeconfig file"
+}
+
+variable "region" {
+  description = "Region where the cluster is created"
+}
+
+variable "resource_group" {
+//  name       = "cloud-pak-sandbox-ibm"
+  description = "Resource group name where the cluster will be hosted."
+}
+
+variable "openshift_version" {
+  default     = "4.6"
+  type        = string
+  description = "Openshift version installed in the cluster"
+}
+//
+//variable "hasEntitlementKey" {
+//  type        = string
+//  description = "Do you have a Cloud Pak for Business Automation Entitlement Registry key (Yes/No, default: No):"
+//}
+
+variable "local_registry_server" {
+  description = "Enter the public image registry or route (e.g., default-route-openshift-image-registry.apps.<hostname>).\nThis is required for docker/podman login validation:"
+}
+
+variable "portworx_is_ready" {
+  type    = any
+  default = null
+}
+
+variable "entitled_registry_key" {
+  type        = string
+//  sensitive = true
+  description = "Do you have a Cloud Pak for Business Automation Entitlement Registry key? If not, Get the entitlement key from https://myibm.ibm.com/products-services/containerlibrary"
+}
+
+variable "namespace" {
+  type        = string
+  description = "namespace for cp4ba"
+}
+
+variable "project_name" {
+  description = "Enter a valid project name. Project name should not be 'openshift' or 'kube' or start with 'openshift' or 'kube'."
+  type = string
+//  validation {
+//    condition = can(regex("^kube+$", var.project_name))
+//    error_message = "Please enter a valid project name that should not be 'openshift' or 'kube' or start with 'openshift' or 'kube'."
+//  }
+}
+
+variable "platform_options" {
+  type        = number
+  description = "Select the cloud platform to deploy. Enter a valid option [1 - 3]: \n 1. RedHat OpenShift Kubernetes Service (ROKS) - Public Cloud \n 2. Openshift Container Platform (OCP) - Private Cloud \n 3. Other ( Certified Kubernetes Cloud Platform / CNCF)"
+}
+
+variable "deployment_type" {
+  type        = number
+  description = "What type of deployment is being performed? Enter a valid option [1 to 2]: \n 1. Demo \n 2. Enterprise"
+}
+
+variable "platform_version" {
+  description = "Enter the platform version"
+}
+
+variable "environment" {
+  default     = "dev"
+  description = "Ignored if `cluster_id` is specified. The environment is combined with `project_name` to name the cluster. The cluster name will be '{project_name}-{environment}-cluster' and all the resources will be tagged with 'env:{environment}'"
+}
+
+variable "use_entitlement" {
```

Comment:
Give a description of what this is.  It should be moved up with the other `roks` variables. It's default should be set to `cloud-pak` here.  This is whether a cloud pak is being installed on the roks cluster.  If so, then the roks cluster cost is greatly reduced. Of not, then the charge is higher.

Reply:
Ok, I got it.